src/mahoji/commands/clue.ts

```diff
@@ -76,21 +113,185 @@ export const clueCommand: OSBMahojiCommand = {
 
 		const randomAddedDuration = randInt(1, 20);
 		duration += (randomAddedDuration * duration) / 100;
+		const poh = await getPOH(user.id);
+		const hasOrnateJewelleryBox = poh.jewellery_box === getPOHObject('Ornate jewellery box').id;
+		const hasJewelleryBox = poh.jewellery_box !== null;
 
-		if (userHasGracefulEquipped(user)) {
-			boosts.push('10% for Graceful');
-			duration *= 0.9;
-		}
+		// Global Boosts
+		const globalBoosts = [
+			{
+				condition: isWeekend,
+				boost: '10% for Weekend',
+				durationMultiplier: 1
+			},
+			{
+				condition: () => user.hasEquippedOrInBank('Max cape'),
+				boost: '10% for Max cape',
+				durationMultiplier: 0.9
+			},
+			{
+				condition: () => !user.hasEquippedOrInBank('Max cape') && user.hasEquippedOrInBank('Construct. cape'),
+				boost: '6% for Construction cape',
+				durationMultiplier: 0.94
+			},
+			{
+				condition: () => hasOrnateJewelleryBox,
+				boost: '10% for Ornate jewellery box',
+				durationMultiplier: 0.9
+			},
+			{
+				condition: () => !hasOrnateJewelleryBox && hasJewelleryBox,
+				boost: '5% for Basic/Fancy jewellery box',
+				durationMultiplier: 0.95
+			}
+		];
 
-		if (isWeekend()) {
-			boosts.push('10% for Weekend');
-			duration *= 0.9;
-		}
+		globalBoosts.forEach(({ condition, boost, durationMultiplier }) => {
+			if (condition()) {
+				boosts.push(boost);
+				duration *= durationMultiplier;
+			}
+		});
 
-		if (user.hasEquippedOrInBank('Achievement diary cape')) {
-			boosts.push('10% for Achievement diary cape');
-			duration *= 0.9;
-		}
+		// Specific boosts
+		const clueTierBoosts: Record<string, ClueBoost[]> = {
```

Comment:
item names hardcoded like this, need to be `getOSItem`, that it will throw an error if you spell the name wrong.